### ISSUE TITLE
binance new unsub method forgets to decrease numSubscriptions

### ISSUE THREAD

**Issue by luyiming** [ccxt/ccxt#23628]
### Programming Languages

Python

### CCXT Version

4.3.94

### Description

I believe the unsub implementation of binance in https://github.com/ccxt/ccxt/pull/23550 has a small bug. 

In the ```watchXXX``` method, e.g., ```watchTradesForSymbols```, calling ```stream (type, streamHash, subParamsLength)``` would return a unique stream index and **increment the numSubscriptions on this stream**. 
https://github.com/ccxt/ccxt/blob/aa6d1b50228794a0819b2445c7831472116eb1c9/ts/src/pro/binance.ts#L1136
https://github.com/ccxt/ccxt/blob/aa6d1b50228794a0819b2445c7831472116eb1c9/ts/src/pro/binance.ts#L194

But the unsub implementation forgets to decrease the numSubscriptions on the stream. After repeated watch and unsub, the numSubscriptions might exceed the limit:

https://github.com/ccxt/ccxt/blob/aa6d1b50228794a0819b2445c7831472116eb1c9/ts/src/pro/binance.ts#L191-L193

### Code

```
  

```

**Comment by carlosmiei** [comment]
Hello @luyiming, you're right! Thanks for letting us know. 

Interestingly, we're still "testing" the unSub methods and didn't announce them publicly,  but seems you're already using them 😅

**Comment by luyiming** [comment]
I have implemented binance unsub in my code for a while using a similar approach. It's glad to see ccxt finally supports it officially.

**Comment by carlosmiei** [comment]
@luyiming Yeah we're starting to add `unWatch` methods. Yesterday we merged `unWatchOHLCVForSymbols` and `unWatchOHLCV`  (will be released shortly)

**Comment by luyiming** [comment]
@carlosmiei I’ve implemented a fix for this https://github.com/ccxt/ccxt/pull/24025. However, I’m not sure if the code style aligns with the project’s conventions. Could you please take a look and let me know if anything needs to be adjusted or improved?
